lib/mail/fields/received_field.rb

```diff
@@ -37,7 +37,9 @@ def info
     end
 
     def formatted_date
-      date_time.strftime("%a, %d %b %Y %H:%M:%S ") + date_time.zone.delete(':')
+      if date_time.respond_to? :strftime and date_time.respond_to? :zone
+        date_time.strftime("%a, %d %b %Y %H:%M:%S ") + date_time.zone.delete(':')
+      end
```

Comment:
Let's add a test case that demonstrates that `#formatted_date` is `nil` for an invalid field 👍

Reply:
Turns out there was a test case for an invalid date in `received_field_spec.rb` but it did not check `#formatted_date`. I verified this was indeed failing on the master branch. Added the test in 3e11433